rhino/src/main/java/org/mozilla/javascript/NullabilityDetectorProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.mozilla.javascript;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+public class NullabilityDetectorProvider {
+    private static NullabilityDetectorProvider provider;
+    private final ServiceLoader<NullabilityDetector> loader;
+
+    private NullabilityDetectorProvider() {
+        loader = ServiceLoader.load(NullabilityDetector.class);
+    }
+
+    public static NullabilityDetectorProvider getInstance() {
+        if (provider == null) {
+            provider = new NullabilityDetectorProvider();
+        }
+
+        return provider;
+    }
+
+    public NullabilityDetector getImpl() {
+        Iterator<NullabilityDetector> iterator = loader.iterator();
```

Comment:
I (very recently) added ScriptRuntime.loadOneServiceImplementation to handle this case and throw an error if there are duplicates (to avoid inconsistency if the classpath is weird). Can you look at using that? It's package-public but could easily be public.